nerfstudio/models/splatfacto.py

```diff
@@ -251,6 +274,93 @@ def populate_modules(self):
         else:
             self.background_color = get_color(self.config.background_color)
 
+        from nerfstudio.viewer.viewer_elements import ViewerButton, ViewerClick, ViewerControl, ViewerDragbox
+
+        self.viewer_control = ViewerControl()
+
+        def delete_cropbox(button: ViewerButton):
+            def on_delete(click: ViewerClick):
+                if not self.selected_gaussian_indices:
+                    return
+                # Assuming self.xys is already updated with the latest projection of Gaussians
+                cumulative_mask = torch.ones(self.xys.shape[0], dtype=torch.bool, device=self.device)
+
+                # Compute a cumulative mask as the intersection of all selected dragbox indices
+                for indices in self.selected_gaussian_indices:
+                    dragbox_mask = torch.zeros(self.xys.shape[0], dtype=torch.bool, device=self.device)
+                    dragbox_mask[list(indices)] = True
+                    # Apply logical AND to accumulate only Gaussians present in all dragboxes
+                    cumulative_mask &= dragbox_mask
+
+                # Invert the cumulative_mask to keep Gaussians outside the intersection
+                keep_mask = ~cumulative_mask
+
+                # Update the Gaussian parameters to keep only the Gaussians outside the intersection
+                for name, param in self.gauss_params.items():
+                    self.gauss_params[name] = torch.nn.Parameter(param[keep_mask])
+                self.remove_from_all_optim(self.optimizers, ~keep_mask)
+                if self.max_2Dsize is not None:
+                    self.max_2Dsize = self.max_2Dsize[keep_mask]
+                    self.xys_grad_norm = self.xys_grad_norm[keep_mask]
+                    self.vis_counts = self.vis_counts[keep_mask]
+
+                # Clear the list of selected dragbox indices after deletion
+                self.selected_gaussian_indices.clear()
+                self.original_colors = None
+
+                self.button2.set_disabled(False)
+                self.viewer_control.unregister_pointer_cb(on_delete)
+
+            # Disable the button and register the callback for dragbox events
+            self.button2.set_disabled(True)
+            self.viewer_control.register_dragbox_cb(on_delete)
+
+        # Create the button with the modified callback
+        self.button2 = ViewerButton("Delete Gaussians", cb_hook=delete_cropbox)
+
+        def select_gaussians(button: ViewerButton):
```

Comment:
Can we make both `select_gaussians` and `delete_cropbox` class functions? Will declutter the populate modules function